lib/image.ts

```diff
@@ -0,0 +1,87 @@
+import * as Bluebird from 'bluebird';
+import * as Docker from 'dockerode';
+import { Readable } from 'stream';
+import * as tar from 'tar-stream';
+import { fs } from 'mz';
+
+export class Image {
+	public static async addLivepushShim(
+		buildContext: Readable,
+		dockerfileRegex = /^(?:\.\/)?Dockerfile$/,
+	): Promise<Readable> {
+		const extract = tar.extract();
+		const pack = tar.pack();
+
+		return new Bluebird<Readable>((resolve, reject) => {
+			extract.on('entry', async (header, stream, next) => {
+				if (dockerfileRegex.test(header.name)) {
+					// We need to process the dockerfile to add the COPY instruction for the shim
+					const dockerfileContent = Image.processDockerfile(
+						(await Image.streamToBuffer(stream)).toString(),
+					);
+					pack.entry(header, dockerfileContent, err => {
+						if (err != null) {
```

Comment:
!==

Reply:
When null testing it's much better to use `!=` as this also checks for undefined.

Reply:
My point was to tell the reader what `err` could possibly be. Looking at `tar-stream`, it can be `null` (coming from `end-of stream`) or `undefined` (coming from `tar-stream`), so you're right. Sorry for nitpicking.

Reply:
No worries, thanks for the review.